lib/improver/cube_combiner.py

```diff
@@ -47,72 +47,274 @@ class CubeCombiner(object):
+                               '*', 'multiply',
                                'max', 'min', 'mean']
         if operation in possible_operations:
             self.operation = operation
         else:
             msg = 'Unknown operation {}'.format(operation)
             raise ValueError(msg)
+        self.warnings_on = warnings_on
 
     def __repr__(self):
         """Represent the configured plugin instance as a string."""
-        desc = '<CubeCombiner: operation={}>'.format(self.operation)
+        desc = ('<CubeCombiner: operation=' +
+                '{}, warnings_on = {}>'.format(self.operation,
+                                               self.warnings_on))
         return desc
 
-    @staticmethod
-    def resolve_metadata_diff(cube1, cube2, revised_coords,
-                              revised_attributes,
-                              warnings_on=False):
-        """Resolve any differences in  metadata between cubes.
+    def resolve_metadata_diff(self, cube1, cube2):
+        """Remove any differences in metadata between cubes.
 
         Args:
             cube1 (iris.cube.Cube):
                 Cube containing data to be combined.
             cube2 (iris.cube.Cube):
                 Cube containing data to be combined.
-            revised_coords (dict or None):
-                Revised coordinates for combined cube.
-            revised_attributes (dict or None):
-                Revised attributes for combined cube.
-            warnings_on (bool):
-                If True output warnings for mismatching metadata.
+
         Returns:
             (tuple): tuple containing
                 **result1** (iris.cube.Cube):
                     Cube with corrected Metadata.
                 **result2** (iris.cube.Cube):
                     Cube with corrected Metadata.
-        """
-        cubes = iris.cube.CubeList([cube1, cube2])
-        unmatching_attributes = compare_attributes(cubes)
-        unmatching_coords = compare_coords(cubes)
 
+        """
         result1 = cube1
         result2 = cube2
+        cubes = iris.cube.CubeList([result1, result2])
+
+        # Processing will be based on cube1 so any unmatching
+        # attributes will be ignored
 
+        # Find mismatching coords
+        unmatching_coords = compare_coords(cubes)
+        # If extra dim coord length 1 on cube1 then add to cube2
+        for coord in unmatching_coords[0]:
+            if coord not in unmatching_coords[1]:
+                if len(result1.coord(coord).points) == 1:
+                    if result1.coord_dims(coord) is not None:
+                        coord_dict = dict()
+                        coord_dict['points'] = result1.coord(coord).points
+                        coord_dict['bounds'] = result1.coord(coord).bounds
+                        coord_dict['units'] = result1.coord(coord).units
+                        coord_dict['metatype'] = 'DimCoord'
+                        self.add_coord(result2, coord, coord_dict)
```

Comment:
I tried this and it didn't always work. Since I simply want this extra dimension to combine the cubes and will then throw the modified cube away I prefer to keep this as is.